tango/executor.py

```diff
@@ -50,13 +52,35 @@ def execute_step_graph(self, step_graph: StepGraph) -> str:
 
         ordered_steps = sorted(step_graph.values(), key=lambda step: step.name)
 
+        # find uncacheable leaf steps
```

Comment:
Turns out that uncacheable leaf steps would never run with the changes I made. That's not right.

Reply:
I thought that was intentional?